### PR TITLE
Completely revamps plasmaman outfit code

### LINKED ISSUE #11315: Plasmamen Security Officers get the wrong mask in their internals box
Gives the normal mask instead of the security one. May affect other roles.

### PR DESCRIPTION
**Pull request by Takahiru**
### Intent of your Pull Request

Before this PR, there were different outfits for every plasmaman job. Problem is, the only differences between these outfits and the regular ones are the uniform and head. Someone didn't bother to make new boxes for the plasmamen either. 

This PR makes it so that the plasmaman outfit, head, and box are simply variables in datum/outfit/job which are equipped if the is a plasmaman.

### Why is this change good for the game?

It also fixes #11315 
Makes it so that changes to outfits also apply to plasmamen, with the exception of the helmet and uniform.

# Changelog
:cl:  
bugfix: Plasmamen now get their respective boxes for their jobs 
tweak: Changes up plasmamen outfit code
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/boxes.dm [hunk 1]
```diff
@@ -141,6 +141,12 @@
 	new /obj/item/crowbar/red(src)
 	new /obj/item/reagent_containers/hypospray/medipen(src)
 
+/obj/item/storage/box/survival_mining/plasmaman/PopulateContents()
```
Comment (alexkar598) [inline code/game/objects/items/storage/boxes.dm:144]: For the survival boxes, why not just call parent, delete all internal tanks in PopulateContents() and add the plasmaman one?

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -233,8 +237,13 @@
 		else
 			back = backpack //Department backpack
 
-	if (isplasmaman(H) && !(visualsOnly)) //this is a plasmaman fix to stop having two boxes
-		box = null
+	if (isplasmaman(H)) //this is a plasmaman fix to stop having two boxes
+		r_hand = /obj/item/tank/internals/plasmaman
```
Comment (alexkar598) [inline code/modules/jobs/job_types/_job.dm]: They are meant to receive a `/obj/item/tank/internals/plasmaman/belt/full`

File: code/game/objects/items/storage/boxes.dm [hunk 3]
```diff
@@ -141,6 +141,12 @@
...
+	new /obj/item/clothing/mask/gas/explorer(src)
+	new /obj/item/tank/internals/plasmaman(src)
```
Comment (alexkar598) [inline code/game/objects/items/storage/boxes.dm]: Meant to receive a `obj/item/tank/internals/plasmaman/belt/full`

File: code/game/objects/items/storage/boxes.dm [hunk 4]
```diff
@@ -162,6 +168,11 @@
 	new /obj/item/tank/internals/emergency_oxygen(src)
 	new /obj/item/reagent_containers/hypospray/medipen(src)
 
+/obj/item/storage/box/security/plasmaman/PopulateContents()
+	new /obj/item/clothing/mask/gas/sechailer(src)
+	new /obj/item/tank/internals/plasmaman(src)
```
Comment (alexkar598) [inline code/game/objects/items/storage/boxes.dm]: Meant to receive a `obj/item/tank/internals/plasmaman/belt/full`

File: code/game/objects/items/storage/boxes.dm [hunk 5]
```diff
@@ -781,7 +792,16 @@
...
 	new /obj/item/reagent_containers/hypospray/medipen(src)
-	
+
+	if(HAS_TRAIT(SSstation, STATION_TRAIT_PREMIUM_INTERNALS))
+		new /obj/item/flashlight/flare(src)
+		new /obj/item/radio/off(src)
+
+/obj/item/storage/box/hug/survival/plasmaman/PopulateContents()
+	new /obj/item/clothing/mask/breath(src)
+	new /obj/item/tank/internals/plasmaman(src)
```
Comment (alexkar598) [inline code/game/objects/items/storage/boxes.dm]: Meant to receive a `obj/item/tank/internals/plasmaman/belt/full`

File: code/modules/jobs/job_types/botanist.dm [hunk 6]
```diff
@@ -38,6 +38,8 @@
 	gloves  =/obj/item/clothing/gloves/botanic_leather
```
Comment (alexkar598) [inline code/modules/jobs/job_types/botanist.dm]:
```suggestion
	gloves = /obj/item/clothing/gloves/botanic_leather
```

File: code/modules/jobs/job_types/warden.dm [hunk 7]
```diff
@@ -54,6 +54,10 @@
 	suit_store = /obj/item/gun/energy/disabler
 	backpack_contents = list(/obj/item/melee/baton/loaded=1) //yogs - ~~added departmental budget ID~~ removes sec budget
 
+	plasmaman_head = /obj/item/clothing/head/helmet/space/plasmaman/security
+	plasmaman_uniform = /obj/item/clothing/under/plasmaman/security
```
Comment (alexkar598) [inline code/modules/jobs/job_types/warden.dm]: They are meant to receive a `/obj/item/clothing/under/plasmaman/security/warden`

File: code/modules/jobs/job_types/medical_doctor.dm [hunk 8]
```diff
@@ -44,6 +44,8 @@
 	l_hand = /obj/item/storage/firstaid/medical
 	suit_store = /obj/item/flashlight/pen
 
+	plasmaman_uniform = /obj/item/clothing/under/plasmaman/medical
+
```
Comment (alexkar598) [inline code/modules/jobs/job_types/medical_doctor.dm:49]: They are meant to also receive a `/obj/item/clothing/head/helmet/space/plasmaman/blue`

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/boxes.dm (hunk 1 above)
Comment (Takahiru) [reply code/game/objects/items/storage/boxes.dm:144]: thanks for looking it over, i believe i made it how you recommend.

**Comment by adamsong** [comment]
The assistant code has not been updated to give plasmaman outfits to assistants, so they now spawn with the regular outfits, and die

**Comment by 0xRai** [comment]
lawyer plasmaman do not spawn with envirosuit on. Happened to me on Round 35501.

**Comment by adamsong** [comment]
This currently has merge conflicts and won't build. Please fix these issues, this PR is stale

**Pull request closed without merging**